src/ray/rpc/gcs_server/gcs_rpc_client.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/rpc/gcs_server/gcs_rpc_client.h"
+
+namespace ray {
+namespace rpc {
+grpc::ChannelArguments GetGcsRpcClientArguments() {
+  grpc::ChannelArguments arguments = CreateDefaultChannelArguments();
+  arguments.SetInt(GRPC_ARG_MAX_MESSAGE_LENGTH, 512 * 1024 * 1024);
+  arguments.SetInt(GRPC_ARG_MAX_RECONNECT_BACKOFF_MS,
+                   ::RayConfig::instance().gcs_grpc_max_reconnect_backoff_ms());
+  arguments.SetInt(GRPC_ARG_MIN_RECONNECT_BACKOFF_MS,
+                   ::RayConfig::instance().gcs_grpc_min_reconnect_backoff_ms());
+  arguments.SetInt(GRPC_ARG_INITIAL_RECONNECT_BACKOFF_MS,
+                   ::RayConfig::instance().gcs_grpc_initial_reconnect_backoff_ms());
+  return arguments;
+}
+
+std::shared_ptr<grpc::Channel> GcsRpcClient::GetDefaultChannel(const std::string &address,
+                                                               int port) {
+  static std::shared_ptr<grpc::Channel> channel_;
+  static grpc::ChannelArguments arguments_;
+  static std::mutex mu_;
+  static std::string address_;
+  static int port_;
+
+  std::lock_guard<std::mutex> guard(mu_);
+  if (channel_ == nullptr) {
+    address_ = address;
+    port_ = port;
+    arguments_ = GetGcsRpcClientArguments();
+    channel_ = BuildChannel(address_, port_, arguments_);
+  }
+  auto arguments = GetGcsRpcClientArguments();
+
+  if (address_ == address && port_ == port &&
+      // Here we only do simple num args check since only when TLS is set, this
+      // could be different.
+      arguments_.c_channel_args().num_args == arguments.c_channel_args().num_args) {
+    return channel_;
+  } else {
+    RAY_LOG(WARNING) << "Generate a new GCS channel: " << address << ":" << port
```

Comment:
When can this happen? When GCS restarts?

Reply:
In testing

Reply:
can you add a comment? (that this should never happen in testing). It might be also great if we use ERROR instead of WARNING and write "This shouldn't happen unless it is testing" in the log message.

Reply:
Hi Sang, I agree with you about this. But this won't impact the correctness. So if there are cases which falls into this and we print, it doesn't look good.

Actually I think if you ray.init to different GCS this might happen in the driver side. Let me follow Chen's comments just update the global client).

Reply:
yeah, that also works. I believe if we have 2 clients at the same time, this could have a correctness issue, but I guess this shouldn't happen.

Reply:
@rkooo567  I think one case:

```
ray.init(GCS_1)
# do something

ray.shutdown()

ray.init(GCS_2)
# do something
ray.shutdown()
```

My concern is ERROR could push the error to the driver which doesn't look good.